### DISCUSSION TITLE
Questions on `doc/admin-deployment.md`

### DISCUSSION THREAD

**Discussion by Mikaela** [EFForg/privacybadger#2947] [Q&A] [answered]
- https://github.com/EFForg/privacybadger/blob/master/doc/admin-deployment.md

I was experimenting with the browser policies for fun and I didn't get the Firefox part working. Are managed storage manifests still a thing?

What did work was `/etc/firefox/policies/policies.json` (https://mozilla.github.io/policy-templates/) although LibreWolf also read it overwriting LibreWolf default policies, so I discourage anyone with LibreWolf trying it:

```json
{
  "policies": {
    "3rdparty": {
      "Extensions": {
        "jid1-MnnxcxisBPnSXQ-eff@jetpack": {
          "showIntroPage": false
        }
      }
    }
}
```

I was also wondering if it's intentional that automatic install instructions are not provided?

```json
{
  "policies": {
    "3rdparty": {
      "Extensions": {
        "jid1-MnnxcxisBPnSXQ-eff@jetpack": {
          "showIntroPage": false
        }
      }
    },
    "ExtensionSettings": {
      "jid1-MnnxcxisBPnSXQ-eff@jetpack": {
        "install_url": "https://www.eff.org/files/privacy-badger-latest.xpi",
        "installation_mode": "normal_installed"
      }
    }
  }
}
```

Chromium version `/etc/opt/chromium/policies/managed/privacy-badger.json`:

```json
{
  "3rdparty": {
    "extensions": {
      "pkehgijcmpdhfbdbbnkijodmdjhbjlgp": {
        "showIntroPage": false
      }
    }
  },
  "ExtensionSettings": {
    "pkehgijcmpdhfbdbbnkijodmdjhbjlgp": {
      "installation_mode": "normal_installed",
      "override_update_url": true,
      "toolbar_pin": "force_pinned",
      "update_url": "https://clients2.google.com/service/update2/crx"
    }
  }
}
```

Chromium seemed to be happy with multiple different json files as long as they didn't contain `3rdparty` or `ExtensionSettings` multiple times. Both of the manifests allow changing `normal_installed` to `force_installed` to prevent uninstallation and the force pin is my personal preference on extensions that may affect functionality of a webpage.

It may also be worth mentioning that the Chromium files work at least for Chromium (`/etc/opt/chromium`), Chrome (`/etc/opt/chrome`) and Microsoft Edge (`/etc/opt/edge`)? Currently the Microsoft Edge mention is only that the same group policy as for Chrome works for it without mentioning that Edge is also available for Linux.

I wasn't able to locate Vivaldi and Brave policy directories (apparently they aren't `/etc/opt/{vivaldi,brave}/policies/managed` or I was doing something wrong), but did test that the Microsoft Edge policy worked for stable, beta and dev channels.

**Comment by ghostwords** [comment, marked as answer]
>Are managed storage manifests still a thing?

Yes, [our doc](https://github.com/EFForg/privacybadger/blob/master/doc/admin-deployment.md) links to Firefox docs for the [managed storage manifests folder](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_manifests#Manifest_location). For Linux:

For global visibility, `/usr/lib/mozilla/managed-storage/<name>.json` or `/usr/lib64/mozilla/managed-storage/<name>.json`

For per-user, `~/.mozilla/managed-storage/<name>.json`.

**Reply by Mikaela** [reply]
Thank you for the answers, I wonder if I should contribute to the file, but I am unsure on where to start with that, but I guess anyone having the issues might be able to find this discussion and hopefully be helped by it

**Comment by ghostwords** [comment]
>I was also wondering if it's intentional that automatic install instructions are not provided?

Our doc contains whatever useful tips came up helping users configure Privacy Badger using admin policies. So far nobody asked for help with automatic installation.

**Comment by ghostwords** [comment]
Thanks for sharing what worked to auto-install Privacy Badger on your Linux device!

**Reply by Mikaela** [reply]
I would also like to share this HTTPS-only mode forcing policy since HTTPS Everywhere extension is no more, the HTTPS-only mode settings don't seem to sync and it still seems too easy to end up on http:// for my taste.

`/etc/opt/chromium/policies/managed/https-everywhere.json`:

```json
{
  "HttpsOnlyMode": "force_enabled",
  "HttpsUpgradesEnabled": true
}
```
